From: Michele Locati <michele@locati.it>
Date: Mon, 7 Sep 2026 15:39:45 +0200
Subject: [PATCH] Fix the PHPDoc of the setters of Url and UrlImmutable

- they accept null (to unset the component), but the parameters are not documented as nullable
- they are documented as returning UrlInterface (self), but Url returns $this and UrlImmutable returns a clone of the same class: with the actual types the subclasses keep their type in method chains

--- a/src/Url.php
+++ b/src/Url.php
@@ -53,6 +53,8 @@ class Url extends AbstractUrl
 
     /**
      * {@inheritdoc}
+     *
+     * @return $this
      */
     public function setScheme($data)
     {
@@ -71,6 +73,8 @@ class Url extends AbstractUrl
 
     /**
      * {@inheritdoc}
+     *
+     * @return $this
      */
     public function setUser($data)
     {
@@ -89,6 +93,8 @@ class Url extends AbstractUrl
 
     /**
      * {@inheritdoc}
+     *
+     * @return $this
      */
     public function setPass($data)
     {
@@ -107,6 +113,8 @@ class Url extends AbstractUrl
 
     /**
      * {@inheritdoc}
+     *
+     * @return $this
      */
     public function setHost($data)
     {
@@ -125,6 +133,8 @@ class Url extends AbstractUrl
 
     /**
      * {@inheritdoc}
+     *
+     * @return $this
      */
     public function setPort($data)
     {
@@ -143,6 +153,8 @@ class Url extends AbstractUrl
 
     /**
      * {@inheritdoc}
+     *
+     * @return $this
      */
     public function setPath($data)
     {
@@ -161,6 +173,8 @@ class Url extends AbstractUrl
 
     /**
      * {@inheritdoc}
+     *
+     * @return $this
      */
     public function setQuery($data)
     {
@@ -179,6 +193,8 @@ class Url extends AbstractUrl
 
     /**
      * {@inheritdoc}
+     *
+     * @return $this
      */
     public function setFragment($data)
     {
--- a/src/UrlInterface.php
+++ b/src/UrlInterface.php
@@ -74,9 +74,9 @@ interface UrlInterface
     /**
      * Set the URL scheme component
      *
-     * @param string $data
+     * @param string|null $data
      *
-     * @return self
+     * @return static
      */
     public function setScheme($data);
 
@@ -90,9 +90,9 @@ interface UrlInterface
     /**
      * Set the URL user component
      *
-     * @param string $data
+     * @param string|null $data
      *
-     * @return self
+     * @return static
      */
     public function setUser($data);
 
@@ -106,9 +106,9 @@ interface UrlInterface
     /**
      * Set the URL pass component
      *
-     * @param string $data
+     * @param string|null $data
      *
-     * @return self
+     * @return static
      */
     public function setPass($data);
 
@@ -122,9 +122,9 @@ interface UrlInterface
     /**
      * Set the URL host component
      *
-     * @param string|array|\Traversable $data
+     * @param string|array|\Traversable|null $data
      *
-     * @return self
+     * @return static
      */
     public function setHost($data);
 
@@ -138,9 +138,9 @@ interface UrlInterface
     /**
      * Set the URL port component
      *
-     * @param string|integer $data
+     * @param string|integer|null $data
      *
-     * @return self
+     * @return static
      */
     public function setPort($data);
 
@@ -154,9 +154,9 @@ interface UrlInterface
     /**
      * Set the URL path component
      *
-     * @param string|array|\Traversable $data
+     * @param string|array|\Traversable|null $data
      *
-     * @return self
+     * @return static
      */
     public function setPath($data);
 
@@ -170,9 +170,9 @@ interface UrlInterface
     /**
      * Set the URL query component
      *
-     * @param string|array|\Traversable $data
+     * @param string|array|\Traversable|null $data
      *
-     * @return self
+     * @return static
      */
     public function setQuery($data);
 
@@ -186,9 +186,9 @@ interface UrlInterface
     /**
      * Set the URL fragment component
      *
-     * @param string $data
+     * @param string|null $data
      *
-     * @return self
+     * @return static
      */
     public function setFragment($data);
 
